src/sinks/datadog/traces/stats.rs

```diff
@@ -0,0 +1,527 @@
+                    version: payload_aggkey.version,
+                    stats: csb,
+                    service: "".to_string(), // not set by the trace-agent, already set in stats
+                    agent_aggregation: "".to_string(), // not set by the trace-agent
+                    sequence: 0,             // not set by the trace-agent
+                    runtime_id: "".to_string(), // not set by the trace-agent
+                    lang: "".to_string(),    //not set by the trace-agent
+                    tracer_version: "".to_string(), // not set by the trace-agent
+                    tags: vec![],            // not set by the trace-agent
+                }
+            })
+            .collect::<Vec<ClientStatsPayload>>()
+    }
+
+    fn export_buckets(&self) -> BTreeMap<PayloadAggregationKey, Vec<ClientStatsBucket>> {
+        let mut m = BTreeMap::<PayloadAggregationKey, Vec<ClientStatsBucket>>::new();
+        self.buckets.values().for_each(|b| {
+            b.export().into_iter().for_each(|(payload_key, csb)| {
+                match m.get_mut(&payload_key) {
+                    None => {
+                        m.insert(payload_key.clone(), vec![csb]);
+                    }
+                    Some(s) => {
+                        s.push(csb);
+                    }
+                };
+            })
+        });
+        m
+    }
+}
+
+fn metric_flag(span: &BTreeMap<String, Value>, key: &str) -> bool {
+    span.get("metrics")
+        .and_then(|m| m.as_object())
+        .map(|m| match m.get(key) {
+            Some(Value::Float(f)) => f.into_inner().signum() == 1.0,
+            _ => false,
+        })
+        .unwrap_or(false)
+}
+
+fn extract_weigth_from_root_span(spans: &[&BTreeMap<String, Value>]) -> f64 {
+    if spans.is_empty() {
+        return 1.0;
+    }
+    let mut parent_id_to_child_weigth = BTreeMap::<i64, f64>::new();
+    let mut span_ids = Vec::<i64>::new();
+    for s in spans.iter() {
+        let parent_id = match s.get("parent_id") {
+            Some(Value::Integer(val)) => *val,
+            _ => 0,
+        };
+        let span_id = match s.get("span_id") {
+            Some(Value::Integer(val)) => *val,
+            _ => 0,
+        };
+        let sample_rate = s
+            .get("metrics")
+            .and_then(|m| m.as_object())
+            .map(|m| match m.get(SAMPLING_RATE_KEY) {
+                Some(Value::Float(v)) => {
+                    let sample_rate = v.into_inner();
+                    if sample_rate <= 0.0 || sample_rate > 1.0 {
+                        1.0
+                    } else {
+                        1.0 / sample_rate
+                    }
+                }
+                _ => 1.0,
+            })
+            .unwrap_or(1.0);
+        if parent_id == 0 {
+            return sample_rate;
+        }
+        span_ids.push(span_id);
+        parent_id_to_child_weigth.insert(parent_id, sample_rate);
```

Comment:
```suggestion
        parent_id_to_child_weight.insert(parent_id, sample_rate);
```

Probably just need a find+replace on `weigth` to `weight` as it's in a few places 😄